plugins/modules/net_tools/nmcli.py

```diff
@@ -1458,6 +1469,18 @@ def tunnel_conn_type(self):
             'sit',
         )
 
+    @staticmethod
+    def normalise_ip4(ip4_addresses):
```

Comment:
```suggestion
    def enforce_ipv4_cidr_notation(ip4_addresses):
```
I think IP normalization typically implies the removal of leading zeroes i.e. `010.010.010.010` becomes `10.10.10.10`.
(Also I think Ansible prefers American English though I didn't find a supporting doc; Python certainly does though)

Reply:
That's not a good idea, it's better to fail on leading zeros. The problem is that leading zeros indicates octal numbers for IP addresses, but only some software interprets it that way - that recently led to severe security issues in various software. (See https://www.cvedetails.com/cve/CVE-2021-29921/ and https://www.bleepingcomputer.com/news/security/python-also-impacted-by-critical-ip-address-validation-vulnerability/, for example.)

Reply:
My suggestion is to not use the name normalization since that's not what these functions do (nor should they).

Reply:
Sorry, was too early for me ;)